src/main/java/com/cburch/logisim/fpga/hdlgenerator/HDL.java

```diff
@@ -79,6 +132,69 @@ public static String xorOperator() {
     return isVHDL() ? " XOR " : "^";
   }
 
+  public static String addOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVHDL() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " + "
+            + typecast(signalTwo, signed)
+            + (isVHDL() ? ")" : "");
+  }
+  
+  public static String subOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVHDL() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " - "
+            + typecast(signalTwo, signed)
+            + (isVHDL() ? ")" : "");
+  }
+
+  public static String shiftlOperator(String signal, int with, int nrOfBits, boolean arithmetic) {
+    if (nrOfBits == 0)
+      return signal;
+    return isVHDL() ? LineBuffer.format("{{1}}{{2}} & {{4}}{{3}}{{4}}", signal, splitVector(with - 1 - nrOfBits, 0), "0".repeat(nrOfBits), nrOfBits == 1 ? "'" : "\"")
```

Comment:
I disagree with @BFH-ktt1 that it is best to describe in VHDL shifts and rotates using bit concatenation. As you remarked @R3dst0ne, the operators `sll`, `srl`, `sla`, `sra`, `rol`, and `ror` are [problematic](https://web.archive.org/web/20160604094455/http://homepage.ntlworld.com/jonathan.deboynepollard/FGA/bit-shifts-in-vhdl.html). However, the functions `shift_left`, `shift_right`, `rotate_left`, and `rotate_right` are fine and work as expected. One has just to be aware that they are only defined for the types `unsigned` and `signed`. So for `std_logic_vector`, two casts are typically required to convey the intent. Also, in synthesis, they yield the expected results achieving shifts/rotates for constant distances using wiring and by variable distances using typically a barrel shifter/rotator.

IMHO, the functions have the advantage that they more clearly convey the intent.

Be aware that the shift rotate distance should be always >= 0! The functions are only defined for `natural` distances!

Reply:
I mark this as resolved, as I agree the style for shift and rotate may get decided in a later PR.